### ISSUE TITLE
exists and findCount queries produce NPE on entity defined with IdClass - SqlTreeProperties.add(SqlTreeProperties.java:38)

### ISSUE THREAD

**Issue by SchlauFuchs** [ebean-orm/ebean#2444]
## Expected behavior
Query is generated and executed
## Actual behavior
Query generation fails with NPE
### Steps to reproduce
create a project with these entities, compile and query
```kotlin
@Entity
class Firm(
    @Id
    val id: String,
    val corporateName: String
)

@Entity
class User(
    @Id
    val id: String,

    @Embedded
    val name: UserName
)

@Embeddable
class UserName(
    val title: String?,
    val givenNames: String,
    val surname: String,
    var preferredName: String?
)

@Entity
@IdClass(value = FirmUserPK::class)
class FirmUserXRef(

    @Id
    @ManyToOne
    val user: User,

    @Id
    @ManyToOne
    val firm: Firm,

    /** if N, the user has been disassociated from the firm */
    val associated: String

)

@Embedded
data class FirmUserPK (
    val user: String?,
    val firm: String?
): Serializable {
    constructor() : this(null, null)
}

@Service
class UserService {
    fun isUserAssociatedWithFirm(userId: String, firmId: String?): Boolean =
        (firmId != null) && QFirmUserXRef()
            .user.id.eq(userId)
            .and().firm.id.eq(firmId)
            .and().associated.eq("Y")
            .exists()
}

@SpringBootTest
internal class UserServiceTest {

    @Autowired
    lateinit var userService: UserService

    @Test
    fun `isUserAssociatedWithFirm returns true for a valid tuple`() {
        assertTrue(userService.isUserAssociatedWithFirm("phaddon", "dghatfield"))
    }
}
```

```console
2021-11-26 14:14:10 ERROR [main] io.ebean.internal - property [_idClass] not found on FirmUserXRef for query - excluding it.

java.lang.NullPointerException
	at io.ebeaninternal.server.query.SqlTreeProperties.add(SqlTreeProperties.java:38)
	at io.ebeaninternal.server.query.SqlTreeBuilder.addProperty(SqlTreeBuilder.java:438)
	at io.ebeaninternal.server.query.SqlTreeBuilder.getBaseSelectPartial(SqlTreeBuilder.java:473)
	at io.ebeaninternal.server.query.SqlTreeBuilder.getBaseSelect(SqlTreeBuilder.java:494)
	at io.ebeaninternal.server.query.SqlTreeBuilder.buildSelectChain(SqlTreeBuilder.java:248)
	at io.ebeaninternal.server.query.SqlTreeBuilder.buildRoot(SqlTreeBuilder.java:209)
	at io.ebeaninternal.server.query.SqlTreeBuilder.build(SqlTreeBuilder.java:112)
	at io.ebeaninternal.server.query.CQueryBuilder.createSqlTree(CQueryBuilder.java:380)
	at io.ebeaninternal.server.query.CQueryBuilder.createSqlTree(CQueryBuilder.java:369)
	at io.ebeaninternal.server.query.CQueryBuilder.buildFetchAttributeQuery(CQueryBuilder.java:197)
	at io.ebeaninternal.server.query.CQueryBuilder.buildFetchIdsQuery(CQueryBuilder.java:215)
	at io.ebeaninternal.server.query.CQueryEngine.findIds(CQueryEngine.java:145)
	at io.ebeaninternal.server.query.DefaultOrmQueryEngine.findIds(DefaultOrmQueryEngine.java:96)
	at io.ebeaninternal.server.core.OrmQueryRequest.findIds(OrmQueryRequest.java:356)
	at io.ebeaninternal.server.core.DefaultServer.exists(DefaultServer.java:1245)
	at io.ebeaninternal.server.querydefn.DefaultOrmQuery.exists(DefaultOrmQuery.java:1423)
	at io.ebean.typequery.TQRootBean.exists(TQRootBean.java:1549)
	at UserService.isUserAssociatedWithFirm(UserService.kt:17)
	at UserServiceTest.isUserAssociatedWithFirm returns false for a valid tuple but the association flag is 'N'(UserServiceTest.kt:21)
	...
	at com.intellij.junit5.JUnit5IdeaTestRunner.startRunnerWithArgs(JUnit5IdeaTestRunner.java:71)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:33)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:235)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:54)
```

**Comment by rbygrave** [comment]
Noting that the `and()`s are not necessary or expected in the query

```java
QFirmUserXRef()
  .user.id.eq(userId)
  .and().firm.id.eq(firmId)
  .and().associated.eq("Y")
  .exists()
```
 instead that should be written as ...  (just removing the `and()`s)
```java
QFirmUserXRef()
  .user.id.eq(userId)
  .firm.id.eq(firmId)
  .associated.eq("Y")
  .exists()
```
... as expressions are added using `AND` by default.

**Comment by SchlauFuchs** [comment]
Noticed. Doesn't help solving the issue though.

**Comment by rbygrave** [comment]
I have reproduce the issue.  Definitely a bug in how the `exists()` query (and `findCount()` query) derive a select clause / select properties.

That is, this issue exists for both `exists()` and `findCount()` queries.  So workarounds are a bit yuk - can use a SqlQuery or findList() for example.

**Comment by SchlauFuchs** [comment]
Thanks, I was doing that  sql query for now :)

**Comment by rbygrave** [comment]
FYI: Just pushed a fix into master, will be included in the next release 12.13.1
